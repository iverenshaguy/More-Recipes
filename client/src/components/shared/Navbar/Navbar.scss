@import '../../../assets/scss/_modules.scss';

.navbar-brand {
  color: white !important;
}

#more-recipes-navbar {
  background-color: rgba(255, 255, 255, 0);
  z-index: 1;
  padding: 1rem 1rem;
  width: 100%;
  -webkit-transition: all 0.3s ease-out;
  -moz-transition: all 0.3s ease-out;
  -o-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;

  #navbar-right {
    float: right;
  }
}

#result-search-form-div {
  display: none;
  background-color: rgba(255, 255, 255, 0.5);

  #search-form {
    #search {
      border: 1px solid $light-green !important;
    }

    .input-group-btn button {
      color: #fff;
      background-color: $light-green;
      border-color: $light-green;
    }
  }
}

.input-group.blur {
  z-index: -1 !important;
}

.search-icon a {
  padding: 0.46rem 0.5rem 0 0;

  &:hover,
  &:focus,
  &:active {
    cursor: pointer;
    color: $green !important;
    text-decoration: none;
  }
}

@media only screen and (min-width: 1200px) {
  #more-recipes-navbar #search-form {
    width: 70%;
  }
}

@media only screen and (min-width: 992px) and (max-width: 1199px) {
  #more-recipes-navbar #search-form {
    width: 60%;
  }
}

@media only screen and (min-width: 768px) and (max-width: 991px) {
  #more-recipes-navbar #search-form {
    width: 50%;
  }
}

@media only screen and (min-width: 576px) and (max-width: 767px) {
  #more-recipes-navbar {
    #search-form {
      display: none !important;
    }

    .search-icon a {
      color: white !important;

      /*padding: 0.5rem 1rem 0 0 ;*/
    }
  }
}

@media only screen and (max-width: 575px) {
  #more-recipes-navbar #search-form {
    display: none;
  }
}

.navbar {
  a.nav-link {
    color: white !important;
  }

  .dropdown-menu {
    position: absolute !important;
    right: 0% !important;
  }

  .btn {
    border: 1px solid white;
    background-color: transparent;
    outline: none;
    color: white;

    &:hover,
    &:focus,
    &:active {
      color: white;
      border: 1px solid $green;
      background-color: $green;
    }
  }

  .fa-heart {
    color: white;
  }

  li.active a {
    /*color: #1B5E20;*/
    border-bottom: 2px solid white !important;
  }
}

/*Navbaar on scroll*/

.affix {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;

  #search {
    border: 1px solid $grey !important;
    border-right: none !important;
    box-shadow: none !important;
  }

  button {
    border: 1px solid $grey !important;
    background-color: transparent;
    outline: none;
    color: #555 !important;
  }

  .search-btn {
    border: 1px solid $grey !important;
    border-left: none !important;
    box-shadow: none !important;
    color: $light-green !important;
  }

  &.navbar li.active a {
    /*color: #1B5E20;*/
    border-bottom: 2px solid $darkgrey !important;
  }
}

#more-recipes-navbar.affix {
  background-color: white;
  box-shadow: 0px 5px 8px 0px rgba(0, 0, 0, 0.1);

  .nav-item > a,
  &.navbar a.nav-link,
  .navbar-brand,
  .fa-heart {
    color: $darkgrey !important;
  }
}

.nav-profile-picture-div {
  height: 2rem;
  width: 2rem;
  text-transform: capitalize;
  border-radius: 50%;

  /*margin-top: -0.06rem;*/

  img {
    height: 2rem;
    width: 2rem;
    text-transform: capitalize;
    border-radius: 50%;
    margin-top: -0.3rem;
  }
}

.nav-profile-picture-div .dropdown-menu {
  width: 20%;
}

.notifications-link .dropdown-menu {
  width: 40% !important;
  left: 58% !important;
}
