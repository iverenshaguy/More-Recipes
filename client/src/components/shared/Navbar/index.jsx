import React from 'react';
import {
  Nav,
  Navbar,
  DropdownMenu,
  DropdownToggle,
  UncontrolledDropdown,
} from 'reactstrap';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import { userPropTypes } from '../../../helpers/proptypes';
import './Navbar.scss';

/**
 * Returns NavbarWrapper Component
 * @export
 * @function NavbarWrapper
 * @param {object} props
 * @returns {component} NavbarWrapper
 */
const NavbarWrapper = (props) => {
  const { user, isAuthenticated } = props;
  const altImage = 'images/user-image-placeholder.png';

  if (!isAuthenticated) {
    return (
      <div>
        <Navbar color="faded" light expand="md" id="more-recipes-navbar">
          <Link to="/" className="navbar-brand">
            MORE RECIPES
          </Link>
        </Navbar>
      </div>
    );
  }

  return (
    <Navbar color="faded" light className="navbar-expand" id="more-recipes-navbar">
      <Link to="/" className="navbar-brand">
        MORE RECIPES
      </Link>
      <div className="collapse navbar-collapse justify-content-between" id="navbarNav">
        <Nav className="ml-auto" navbar>
          <UncontrolledDropdown nav className="justify-content-start notifications-link">
            <DropdownToggle nav>
              <FontAwesome name="bell" size="lg" />
            </DropdownToggle>
            <DropdownMenu right>
              <div className="container-fluid">
                <p>You have a new review</p>
                <div className="dropdown-divider" />
                <p>3 people viewed your recipe</p>
                <div className="dropdown-divider" />
                <p>Review your security details</p>
                <div className="dropdown-divider" />
                <p>Chef Steve just posted a new recipe</p>
                <div className="dropdown-divider" />
                <p>1 person added your recipe to their favorites</p>
              </div>
            </DropdownMenu>
          </UncontrolledDropdown>
          <UncontrolledDropdown nav>
            <div className="nav-profile-picture-div rounded-cirle menu-dropdown d-inline">
              <DropdownToggle nav>
                <img
                  src={user && user.profilePic ? user.profilePic : altImage}
                  onError={(e) => { e.target.src = altImage; }}
                  className="rounded-cirle"
                  alt="User"
                />
              </DropdownToggle>
              <DropdownMenu right>
                <Link to={`/${user && user.username}`} className="dropdown-item">
                  Profile
                </Link>
                <Link to={`/${user && user.username}/profile/edit`} className="dropdown-item">Edit Profile</Link>
                {/* <div className="dropdown-divider" /> */}
                {/* <a className="dropdown-item" id="my-recipes" href="user-profile.html">
                    My Recipes
                  </a> */}
                {/* <div className="dropdown-divider" /> */}
                {/* <a className="dropdown-item">Settings</a> */}
                <div className="dropdown-divider" />
                <a className="dropdown-item" href="/login" onClick={props.logout}>
                  Log out
                </a>
              </DropdownMenu>
            </div>
          </UncontrolledDropdown>
          <UncontrolledDropdown nav>
            <DropdownToggle nav className="menu-dropdown">
              <FontAwesome name="reorder" size="lg" />
            </DropdownToggle>
            <DropdownMenu right>
              <Link
                to="/recipes/new"
                className="dropdown-item add-recipe"
                href="/recipes/new"
                title="New Recipe"
              >
                Add a Recipe
              </Link>
              {/* <a className="dropdown-item get-favorites" href="user-profile.html">
                  Favorite Recipes
                </a> */}
              <Link to="/" className="dropdown-item">
                Top Recipes
              </Link>
              <a
                className="dropdown-item"
                href="#favorite-recipes"
                title="favorite-recipes"
              >
                Favorite Recipes
              </a>
            </DropdownMenu>
          </UncontrolledDropdown>
        </Nav>
      </div>
    </Navbar>
  );
};

NavbarWrapper.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
  logout: PropTypes.func.isRequired,
  showAddRecipeModal: PropTypes.func.isRequired,
  ...userPropTypes
};

export default NavbarWrapper;
