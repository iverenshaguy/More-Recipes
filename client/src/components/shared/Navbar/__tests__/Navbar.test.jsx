import React from 'react';
import Navbar from '../index';

const logoutMock = jest.fn();

const authProps = {
  isAuthenticated: false,
  type: 'auth',
  logout: logoutMock
};

const normalProps = {
  isAuthenticated: true,
  type: 'home',
  logout: logoutMock
};

describe('Navbar', () => {
  it('renders correctly when app is not authenticated', () => {
    const wrapper = shallow(<Navbar {...authProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly when type = home', () => {
    const wrapper = shallow(<Navbar {...normalProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('logs user out on logout click', () => {
    const wrapper = shallow(<Navbar {...normalProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.find('a[href="/login"]').simulate('click', { preventDefault() { } });

    expect(logoutMock).toHaveBeenCalled();
  });
});
