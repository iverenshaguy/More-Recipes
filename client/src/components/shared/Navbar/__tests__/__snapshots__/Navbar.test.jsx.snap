// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly when authenticated 1`] = `
<Navbar
  className="navbar-expand"
  color="faded"
  expand={false}
  id="more-recipes-navbar"
  light={true}
  tag="nav"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="navbar-brand"
    to="/"
  >
    MORE RECIPES
  </NavLink>
  <div
    className="collapse navbar-collapse justify-content-between"
    id="navbarNav"
  >
    <Nav
      className="ml-auto"
      navbar={true}
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-link"
          to="/"
        >
          <FontAwesome
            name="home"
          />
        </NavLink>
      </NavItem>
      <UncontrolledDropdown
        className="justify-content-start notifications-link"
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="bell"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <div
            className="container-fluid"
          >
            <p>
              You have a new review
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              3 people viewed your recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Review your security details
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              Chef Steve just posted a new recipe
            </p>
            <div
              className="dropdown-divider"
            />
            <p>
              1 person added your recipe to their favorites
            </p>
          </div>
        </DropdownMenu>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <div
          className="nav-profile-picture-div rounded-cirle menu-dropdown d-inline"
        >
          <DropdownToggle
            aria-haspopup={true}
            color="secondary"
            nav={true}
          >
            <img
              alt="User"
              className="rounded-cirle"
              src="images/profile/user1.jpg"
            />
             Jane
          </DropdownToggle>
          <DropdownMenu
            flip={true}
            right={true}
            tag="div"
          >
            <a
              className="dropdown-item"
              href="user-profile.html"
            >
              Profile
            </a>
            <a
              className="dropdown-item"
            >
              Edit Profile
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="user-profile.html"
            >
              My Recipes
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
            >
              Settings
            </a>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
            >
              Log out
            </a>
          </DropdownMenu>
        </div>
      </UncontrolledDropdown>
      <UncontrolledDropdown
        nav={true}
      >
        <DropdownToggle
          aria-haspopup={true}
          className="menu-dropdown"
          color="secondary"
          nav={true}
        >
          <FontAwesome
            name="reorder"
            size="lg"
          />
        </DropdownToggle>
        <DropdownMenu
          flip={true}
          right={true}
          tag="div"
        >
          <a
            className="dropdown-item add-recipe"
            data-toggle="modal"
            href="#add-edit-modal"
            title="New Recipe"
          >
            Add a Recipe
          </a>
          <a
            className="dropdown-item"
            href="search-recipes-user.html"
          >
            All Recipes
          </a>
          <a
            className="dropdown-item get-favorites"
            href="user-profile.html"
          >
            Favorite Recipes
          </a>
          <a
            className="dropdown-item"
            href="top-recipes-user.html"
          >
            Top Recipes
          </a>
        </DropdownMenu>
      </UncontrolledDropdown>
    </Nav>
  </div>
</Navbar>
`;

exports[`Home renders correctly when not authenticated 1`] = `
<div>
  <Navbar
    color="faded"
    expand="md"
    id="more-recipes-navbar"
    light={true}
    tag="nav"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      MORE RECIPES
    </NavLink>
    <div
      className="collapse navbar-collapse justify-content-between"
      id="navbarNav"
    >
      <Nav
        className="ml-auto"
        navbar={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <a>
            <Button
              color="outline-success"
              tag="button"
            >
              SIGNIN
            </Button>
          </a>
        </NavItem>
      </Nav>
    </div>
  </Navbar>
</div>
`;
